src/components/gr-button/gr-button.ts

```diff
@@ -1,55 +1,129 @@
 import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
-import { customElement, property, state } from 'lit/decorators.js';
+import { customElement, property } from 'lit/decorators.js';
 import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
 import style from './gr-button.css';
 
-export type ButtonVariant = 'primary' | 'secondary';
-
 /**
  * @tag gr-button
  * @summary Grace Button component
  *
  * @property {boolean} disabled - Disables the button
- * @property {boolean} medium - Sets size to medium
- * @property {boolean} small - Sets size to small
- * @property {string} variant - Sets the button variant
+ * @property {boolean} primary - Sets size to primary
+ * @property {boolean} secondary - Sets size to secondary
+ * @property {boolean} tertiary - Sets size to tertiary
+ * @property {boolean} success - Sets size to success
+ * @property {boolean} error - Sets size to error
+ * @property {boolean} outline - Sets button outline version
+ * @property {string} size - Sets the button size
+ * @property {string} link - Sets the button tag, either anchor or button
+ * @property {boolean} href - Sets button type to link button
+ * @property {boolean} target - Sets button target (should be defined with href)
+ *
+ * @cssproperty --gr-button-display - Sets the display property of button. Default value is 'inline-flex'.
  *
- * @cssproperty --gr-button-size - Sets the height of button. 40px or 32px
+ * @event {CustomEvent} bl-click
  *
- * @event {CustomEvent} gr-blur - dsdasda
  *
  */
+
+export type ButtonSize = 'small' | 'medium' | 'large';
+export type TargetType = '_blank' | '_parent' | '_self' | '_top';
+
 @customElement('gr-button')
 export default class GrButton extends LitElement {
   static get styles(): CSSResultGroup {
     return [style];
   }
 
-  @state() private hasFocus = false;
+  @property({ type: String, attribute: 'size' })
```

Comment:
Do we need `attribute` option here?

Also since we use size attribute selector in our styles, we need to use `reflect: true` to be sure that we’ll always have `size` attribute even if property set via JS.

Same applies for other attributes that we use in CSS (like `secondary`, `success`, `link`, `outline` etc.) We need `@property` definitions for them to make it possible to set them as component DOM property, not just as an HTML attribute.

Reply:
That's an interesting point. I will look at it.